### PR TITLE
refactor: use contextlib.suppress instead of except pass

### PR DESCRIPTION
**Pull request by kkirsche**
# Pull Request Check List

Resolves: N/A

- [X] Added **tests** for changed code.
- [X] Updated **documentation** for changed code.

This merge request converts `try` / `except` blocks with no `except` action to use `contextlib.suppress` to silence a specific error, instead of passing in an exception handler.

#### Example Before

```python
try:
    travel_world(days=80)
except DistractionError:
    pass
```

#### Example After

```python
import contextlib

with contextlib.suppress(DistractionError):
    travel_world(days=80)
```

#### Explanation
The context manager slightly shortens the code and significantly clarifies the author's intention to ignore the specific errors. The standard library feature was introduced following a discussion, where the consensus was that

> A key benefit here is in the priming effect for readers... The with statement form makes it clear before you start reading the code that certain exceptions won't propagate.

This ignores a few locations that use `try` / `except` / `else`

### CODE REVIEW

File: get-poetry.py [hunk 1]
```diff
@@ -30,7 +30,7 @@
 import tempfile
 
 from contextlib import closing
-from contextlib import contextmanager
+from contextlib import contextmanager, suppress
```
Comment (branchvincent) [inline get-poetry.py:33]: this install script is deprecated, but still technically supports python2. let's revert this change, but the rest look good

**Review by branchvincent — CHANGES_REQUESTED** [review]

File: get-poetry.py (hunk 1 above)
Comment (neersighted) [reply get-poetry.py:33]: To clarify further, we're considering it frozen and not subject to modification outside critical defects like a security issue. Even code formatting is not being applied.

**Review by neersighted — CHANGES_REQUESTED** [review]
I'd rather import all of contextlib and invoke the context manager as `contextlib.suppress` unless you see a compelling reason not to -- it's more explicit given that `suppress` is not a common short import name.

**Comment by kkirsche** [comment]
Including feedback from @neersighted in 32c34884a63b3fd2ca3e90f867b6b364ca33a4fc
. If there is a reason to revert this, we can use git revert accordingly 👍  hopefully this will help ensure that your time is respected reviewing this.

**Review by branchvincent — APPROVED** [review]
@kkirsche thank you!

**Comment by Callek** [comment]
@branchvincent @kkirsche - IMO, Probably shouldn't have touched the pyyaml and ansible `setup.py` files in the test folder I don't think.

**Comment by kkirsche** [comment]
Why, @Callek?

**Comment by kkirsche** [comment]
I'll submit a similar PR upstream to normalize them, if applicable to their supported versions

**Comment by Callek** [comment]
> Why, @Callek?
> 
> EDIT: Oh, it looks like you're vendoring them. Didn't realize that, sorry. I thought they were just local forks

Yea, because they are vendored -- but I should be super explicit, I'm not a maintainer on poetry nor on the two projects I mentioned there. So my comments are merely as a passerby.

**Comment by neersighted** [comment]
> @branchvincent @kkirsche - IMO, Probably shouldn't have touched the pyyaml and ansible `setup.py` files in the test folder I don't think.

Good catch, thanks!
